src/azure-cli-core/azure/cli/core/telemetry_upload.py

```diff
@@ -82,25 +77,30 @@ def upload(data_to_save):
             sys.stdout.write('ERROR: {}/n'.format(str(err)))
             sys.stdout.write('Raw [{}]/n'.format(data_to_save))
 
-    for record in data_to_save:
-        name = record['name']
-        raw_properties = record['properties']
-        properties = {}
-        measurements = {}
-        for k in raw_properties:
-            v = raw_properties[k]
-            if isinstance(v, six.string_types):
-                properties[k] = v
-            else:
-                measurements[k] = v
-        client.track_event(record['name'], properties, measurements)
-
-        if in_diagnostic_mode():
-            sys.stdout.write(
-                '\nTrack Event: {}\nProperties: {}\nMeasurements: {}'.format(name, json.dumps(properties, indent=2),
-                                                                             json.dumps(measurements, indent=2)))
-
-    client.flush()
+    for instrumentation_key in data_to_save:
+        client = TelemetryClient(instrumentation_key=instrumentation_key,
+                                 telemetry_channel=TelemetryChannel(queue=SynchronousQueue(LimitedRetrySender())))
+        enable(instrumentation_key)
+
+        for record in data_to_save[instrumentation_key]:
+            name = record['name']
+            raw_properties = record['properties']
+            properties = {}
+            measurements = {}
+            for k in raw_properties:
+                v = raw_properties[k]
```

Comment:
You can do: `for k, v in raw_properties.items():`

Original code can be improved.